engine/src/main/java/org/terasology/rendering/assets/texture/JPEGDecoder.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.rendering.assets.texture;
+
+import java.nio.ByteBuffer;
+import java.io.InputStream;
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.awt.image.DataBuffer;
+import java.awt.image.DataBufferByte;
+import java.awt.image.WritableRaster;
+
+/**
+ * @author Viveret (github.com/viveret)
+ */
+public class JPEGDecoder {
+    public static enum Format {
+        ERR, RGB, RGBA
```

Comment:
These are for output formats, if you wanted to read in a JPG with 4 components for some odd reason. If this should not be a case then it could be removed.